apps/ledger-live-desktop/src/renderer/screens/market/MarketRowItem.tsx

```diff
@@ -228,11 +245,23 @@ function MarketRowItem({
                   <Button
                     data-test-id={`market-${currency?.ticker}-swap-button`}
                     variant="color"
+                    mr={1}
                     onClick={onSwap}
                   >
                     {t("accounts.contextMenu.swap")}
                   </Button>
                 )}
+                {availableOnStake && (
+                  <Box onClick={e => onStake(e)}>
+                    <Button
+                      data-test-id={`market-${currency?.ticker}-stake-button`}
+                      variant="color"
+                      onClick={stakeFlow({ currencies: [currency?.id] })}
```

Comment:
this is effectively calling a hook within a callback - which react recommends you don't do. If `stakeFlow` was called `useStakeFlow` I think this will fail linting with an error like:

```
React Hook "useStakeFlow" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function. (react-hooks/rules-of-hooks)eslint
```

Reply:
If I use a callback, we will have the "hook" issue... I do not find a better solution.

Reply:
I think we can make the following work?

```
const startStakeFlow = useStakeFlow();

return <button onClick={() => startStateFlow({currencies: [currency?.id]) />}
```

Also, don't we want to show the account selection drawer when they click on this button? They don't need to select a single currency and then the account.